src/elf/reloc.rs

```diff
@@ -391,4 +380,111 @@ if_alloc! {
             }
         }
     }
+
+    #[derive(Default)]
+    pub struct RelocSection<'a> {
+        bytes: &'a [u8],
+        count: usize,
+        ctx: RelocCtx,
+        start: usize,
+        end: usize,
+    }
+
+    impl<'a> fmt::Debug for RelocSection<'a> {
+        fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+            let len = self.bytes.len();
+            fmt.debug_struct("RelocSection")
+                .field("bytes", &len)
+                .field("range", &format!("{:#x}..{:#x}", self.start, self.end))
+                .field("count", &self.count)
+                .field("Relocations", &self.to_vec())
+                .finish()
+        }
+    }
+
+    impl<'a> RelocSection<'a> {
+        #[cfg(feature = "endian_fd")]
+        pub fn parse(bytes: &'a [u8], offset: usize, filesz: usize, is_rela: bool, ctx: Ctx) -> ::error::Result<RelocSection<'a>> {
+            // TODO: better error message when too large (see symtab implementation)
+            let bytes = bytes.pread_with(offset, filesz)?;
+
+            Ok(RelocSection {
+                bytes: bytes,
+                count: filesz / Reloc::size(is_rela, ctx),
+                ctx: (is_rela, ctx),
+                start: offset,
+                end: offset + filesz,
+            })
+        }
+
+        /// Try to parse a single relocation from the binary, at `index`.
+        #[inline]
+        pub fn get(&self, index: usize) -> Option<Reloc> {
+            if index >= self.count {
+                None
+            } else {
+                Some(self.bytes.pread_with(index * Reloc::size_with(&self.ctx), self.ctx).unwrap())
```

Comment:
Because constructor checks the size, we're safe to unwrap here and in the iterator, correct?

Reply:
Yep, pretty much.